### PR TITLE
[FIX] account: Negative fixed taxes on 100% discount

### PR DESCRIPTION
**Pull request by nboulif**
Taxes calculation improved so must revert the following commit : https://github.com/odoo/odoo/pull/53788/commits/8c50e4772d9c5c5758e83302a001cc6f5c7af284

opw-2378978

### CODE REVIEW

**Comment by nboulif** [comment]
@simongoffin

**Comment by smetl** [comment]
@nboulif This bug is still there whatever you change this condition or not.  I think we should do something like:
```
if base < 0 or (currency.is_zero(base) and price_unit * quantity < 0):
```

**Comment by nboulif** [comment]
Enterprise PR : https://github.com/odoo/enterprise/pull/15035

File: addons/account/models/account_move.py [hunk 1]
```diff
@@ -2985,7 +2993,11 @@ def _get_fields_onchange_balance_model(self, quantity, discount, balance, move_t
             # 220           | 10% incl, 5%  |                   | 200               | 230
             # 20            |               | 10% incl          | 20                | 20
             # 10            |               | 5%                | 10                | 10
-            taxes_res = taxes._origin.compute_all(balance, currency=currency, handle_price_include=False)
+            if move_type in ('out_invoice', 'in_refund', 'out_receipt'):
+                force_sign = -1
+            else:
+                force_sign = 1
```
Comment (smetl) [inline addons/account/models/account_move.py:2999]: create an helper for this plz :p

Comment (nboulif) [reply addons/account/models/account_move.py]: Done :+1:

**Comment by smetl** [comment]
@nboulif branch is ok but red ;(

File: addons/account/models/account_move.py [hunk 2]
```diff
@@ -2865,7 +2876,7 @@ def _get_price_total_and_subtotal_model(self, price_unit, quantity, discount, cu
 
         # Compute 'price_total'.
         if taxes:
-            taxes_res = taxes._origin.compute_all(price_unit_wo_discount,
+            taxes_res = taxes._origin.with_context(force_sign=self._get_tax_force_sign()).compute_all(price_unit_wo_discount,
```
Comment (smetl) [inline addons/account/models/account_move.py:2879]: arf you can't do that here but you have `move_type` as parameter

**Comment by smetl** [comment]
@robodoo r+

**Comment by nboulif** [comment]
@fw-bot ignore

**Comment by nboulif** [comment]
@fw-bot up to master

**Pull request closed without merging**